h2o-core/src/test/java/water/fvec/VecShuffleTest.java

```diff
@@ -0,0 +1,243 @@
+                }
+            }
+            seen = false;
+        }
+        return elementMissing;
+    }
+
+    /**
+     * Compare values of Vec and returns the randomness change 
+     * (0.0 same vector, 1.0 perfect shuffle)
+     */
+    double randomnessCoefficient(Vec og, Vec nw) {
+        int changedPlaces = 0;
+        switch (og.get_type()) {
+            case Vec.T_STR:
+                for (long i = 0; i < nw.length(); ++i) {
+                    if (!og.stringAt(i).equals(nw.stringAt(i)))
+                        changedPlaces++;
+                }
+                break;
+            case Vec.T_UUID:
+                for (long i = 0; i < nw.length(); ++i) {
+                    if (og.at16l(i) != nw.at16l(i))
+                        changedPlaces++;
+                }
+                break;
+            case Vec.T_NUM: /* fallthrough */
+            case Vec.T_CAT:
+            case Vec.T_TIME:
+            default:
+                
+                for (long i = 0; i < nw.length(); ++i) {
+                    double l = og.at(i);
+                    double r = nw.at(i);
+                    if (og.at(i) != nw.at(i))
+                        changedPlaces++;
+                }
+                break;
+        }
+        return (changedPlaces * 1.0) / nw.length();
+    }
+
+
+    /**
+     * Testing algorithm multiple times containing Int and String
+     */
+    @Test
+    public void shuffleSmallFrame(){
+        Frame fr = null;
+        try {
+            Scope.enter();
+            fr = new TestFrameBuilder()
+                    .withVecTypes(Vec.T_NUM, Vec.T_STR)
+                    .withColNames("first", "second")
+                    .withDataForCol(0, ard(1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0))
+                    .withDataForCol(1, ar("a", "b", "c", "d", "e", "f","g", "h", "i", "j", "k"))
+                    .build();
+            Scope.track(fr);
+            DKV.put(fr._key, fr);
+
+            for (int i = 0 ; i < 10 ; i++) {
+                for (int j = 0; j < 2 ; j++){
+                    // Get shuffled Vec
+                    Vec shuffledFeature = VecUtils.ShuffleVec(fr.vec(j), fr.vec(j).makeCopy());
+                    double rndCoe = randomnessCoefficient(fr.vec(j), shuffledFeature);
+                    Assert.assertEquals(1.0, rndCoe, 1e-2);
+                    Assert.assertFalse(checkElements(fr.vec(j), shuffledFeature));
+                }
+            }
+        } finally {
+            Scope.exit();
+            if (fr != null) fr.remove();
+        }
+    }
+
+    /**
+     * Test Bigger frame with unique Double values
+     */
+    @Test
+    public void shuffleBigVec(){
+        Key parsed = Key.make("shuffle_test_2_parsed");
+        Frame fr = null;
+        Vec shuffledFeature = null ;
+        try {
+            Scope.enter();
+           // contains 10,000,000 unique doubles
+            fr = parse_test_file(parsed, "smalldata/jira/shuffle_test_2.csv");
```

Comment:
I see, I will use the bigdata file (loans) for this test and remove the files I have added to the smalldata folder.

Reply:
I see, I will use the bigdata file (loans) for this test and remove the files I have added to the smalldata folder. Was using this one because every row has a unique value (to test the performance of the shuffling algorithm).